onnxruntime/core/providers/vitisai/vitisai_execution_provider.cc

```diff
@@ -2,22 +2,80 @@
 // Licensed under the MIT License.
 #include "vitisai_execution_provider.h"
 
+// Standard headers/libs.
 #include <cassert>
 #include <fstream>
 #include <istream>
+#include <filesystem>
 
 #include "vaip/capability.h"
 #include "vaip/global_api.h"
+#include "ep_context_utils.h"
 
 using namespace ONNX_NAMESPACE;
 
+namespace fs = std::filesystem;
+
 namespace onnxruntime {
 constexpr const char* VITISAI = "VITISAI";
 
 VitisAIExecutionProvider::VitisAIExecutionProvider(
     const ProviderOptions& info)
+    // const ProviderOptions& info, const SessionOptions* p_sess_opts)
     : IExecutionProvider{onnxruntime::kVitisAIExecutionProvider}, info_(info) {
   CreateKernelRegistry();
+
+#if 0
+  if (p_sess_opts) {
```

Comment:
Still some pieces missing? e.g how to pass in the SessionOptions, debugging code in #if 0

Reply:
https://github.com/microsoft/onnxruntime/blob/98d7208bfe8b697d3ac25960b3f355b2ce2a86fe/onnxruntime/core/session/provider_bridge_ort.cc#L2821-L2824

```
  // EP context related session config options.
  provider_options["ep_context_enable"] = options->value.config_options.GetConfigOrDefault(kOrtSessionOptionEpContextEnable, "0");
  provider_options["ep_context_embed_mode"] = options->value.config_options.GetConfigOrDefault(kOrtSessionOptionEpContextEmbedMode, "1");
  provider_options["ep_context_file_path"] = options->value.config_options.GetConfigOrDefault(kOrtSessionOptionEpContextFilePath, "");
```

Reply:
OK, find the code in SessionOptionsAppendExecutionProvider_VitisAI which copies the settings from session options to provider option.
Need to remove these debugging code.

Reply:
Will do.